CI/unit_tests/utils/test_molecule_class.py

```diff
@@ -0,0 +1,63 @@
+"""
+MDSuite: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Unit tests for the molecule data class.
+"""
+import pytest
+
+import mdsuite
+from mdsuite.utils.molecule import Molecule
+
+
+def test_instantiation():
+    """
+    Test that the class is instantiated correctly.
+
+    Notes
+    -----
+    Test the following cases:
+
+    * Fail when essential data is not provided.
+    * Store the correct information when provided.
+    * Use the correct defaults.
+    """
+    with pytest.raises(TypeError):
+        Molecule()
+        Molecule(name="test")
+        Molecule(name="test", amount=2)
+        Molecule(name="test", cutoff=4)
```

Comment:
Actually I think we can remove this entire test. If we don't have this test coverage tells more precisely which parts of this dataclass are used. E.g. if we removed smiles it would still be covered in this test even though it is not used and we would not know.

Reply:
Sure I can remove it. The idea was more that in my construction of this data-class somethings are defaulted to None while others are essential. This just ensures that if some of the essential data is not parsed it fails. In the case where someone set a cutoff to a default this would fail which would be a change on our end.

Reply:
I vote to remove it. @christophlohrmann

Reply:
It does serve a purpose as a test. If any of those parameters is made default it will fail. I can add a line that makes sure if another is made non-optional it also fails. But this test is not testing the function of a data-class rather testing things that are default are left that way

Reply:
So the test here is to ensure that `name, amount, cutoff` aren't set to default values in the `mdsuite.Molecule` ?
In that case where would that cause failure in the code? Can you change the test to test for the failure in the code rather than checking that it is not changed?

Reply:
That would fail in the code as the user defines a Molecule directly in a script.

Reply:
I am leaving this for a vote from @christophlohrmann upon which we can discuss what the best approach is here. The way it is tested is exactly how a user would use it in a script. It is ensuring that a user cannot enter insufficient information into a class.

Reply:
@PythonFZ I would like to make sure you are comfortable with this. The way the code is tested here is exactly how it is called in the real use cases. If you are very unhappy with it, perhaps we can discuss that further.